indexer/services/ender/__tests__/lib/kafka-publisher.test.ts

```diff
@@ -139,11 +169,247 @@ describe('kafka-publisher', () => {
+      type: FillType.MARKET,
+      clobPairId: '1',
+      orderId: testConstants.defaultOrderId,
+      size: '10',
+      price: '20000',
+      quoteAmount: '200000',
+      eventId: testConstants.defaultTendermintEventId,
+      transactionHash: '', // TODO: Add a real transaction Hash
+      createdAt: testConstants.createdDateTime.toISO(),
+      createdAtHeight: testConstants.createdHeight,
+      clientMetadata: '0',
+      fee: '1.1',
+    };
+    const order: OrderFromDatabase = {
+      ...testConstants.defaultOrderGoodTilBlockTime,
+      id: testConstants.defaultOrderId,
+    };
+
+    const recipientSubaccountId: IndexerSubaccountId = IndexerSubaccountId.fromPartial({
+      owner: 'recipient',
+      number: 1,
+    });
+    const deposit: TransferFromDatabase = {
+      id: '',
+      senderWalletAddress: defaultWalletAddress,
+      recipientSubaccountId: SubaccountTable.uuid(
+        recipientSubaccountId.owner,
+        recipientSubaccountId.number,
+      ),
+      assetId: testConstants.defaultAsset.id,
+      size: '10',
+      eventId: testConstants.defaultTendermintEventId,
+      transactionHash: 'hash',
+      createdAt: DateTime.utc().toISO(),
+      createdAtHeight: '1',
+    };
+    it('successfully aggregates all fill events per order id and sorts messages', async () => {
+      const publisher: KafkaPublisher = new KafkaPublisher();
+
+      // merged with message 3.
+      const msg1Contents: SubaccountMessageContents = {
+        fills: [
+          generateFillSubaccountMessage(fill, 'BTC-USD'),
+        ],
+      };
+      const message1: AnnotatedSubaccountMessage = {
+        blockHeight: '1',
+        transactionIndex: 1,
+        eventIndex: 1,
+        contents: JSON.stringify(msg1Contents),
+        subaccountId: {
+          owner: 'owner1',
+          number: 0,
+        },
+        version: '1',
+        orderId: 'order1',
+        isFill: true,
+      };
+
+      const msg2Contents: SubaccountMessageContents = {
+        fills: [
+          generateFillSubaccountMessage(fill, 'ETH-USD'),
+        ],
+      };
+      const message2: AnnotatedSubaccountMessage = {
+        blockHeight: '1',
+        transactionIndex: 2,
+        eventIndex: 2,
+        contents: JSON.stringify(msg2Contents),
+        subaccountId: {
+          owner: 'owner2',
+          number: 0,
+        },
+        version: '1',
+        orderId: 'order2',
+        isFill: true,
+      };
+
+      const msg3Contents: SubaccountMessageContents = {
+        fills: [
+          generateFillSubaccountMessage({
+            ...fill,
+            size: '100',
+          }, 'BTC-USD'),
+        ],
+        orders: [
+          generateOrderSubaccountMessage(order, 'BTC-USD'),
+        ],
+      };
+      const message3: AnnotatedSubaccountMessage = {
+        blockHeight: '1',
+        transactionIndex: 3,
+        eventIndex: 3,
+        contents: JSON.stringify(msg3Contents),
+        subaccountId: {
+          owner: 'owner3',
```

Comment:
nit: For the sake of correct-ness, this should match msg1.

Reply:
done